src/inc/volatile.h

```diff
@@ -120,7 +120,7 @@ T VolatileLoad(T const * pt)
     STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
 
 #ifndef DACCESS_COMPILE
-#if defined(_ARM64_) && defined(__GNUC__)
+#if defined(_ARM64_) && defined(__GNUC__) && !(defined(__BIONIC__))
```

Comment:
@cydhaselton but this error doesn't mean that the __atomic_load was not found. It means that the first parameter should not be `volatile int*`, but rather `int *`. The signature of the function is `type __atomic_load(const type* atomic_obj, int mem_ord);`
So the cast at line 128 is incorrect, removing "volatile" should fix it. And I think that then you can drop the change at line 123.